propagator/opentelemetry-propagator-b3/src/opentelemetry/propagators/b3/__init__.py

```diff
@@ -94,18 +92,15 @@ def extract(
             )
 
         if (
-            self._trace_id_regex.fullmatch(trace_id) is None
+            trace_id == trace.INVALID_TRACE_ID
+            or span_id == trace.INVALID_SPAN_ID
+            or self._trace_id_regex.fullmatch(trace_id) is None
```

Comment:
When `trace_id` or `span_id` are "invalid" further expressions are not evaluated, because Python `or` is a short circuit operator: https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not. Also implemented unit tests cover the mention case without throwing any error.
Considering another code  branch (there is a header and extraction happens) - getter in its interface guarantees optional string: https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-api/src/opentelemetry/propagators/textmap.py#L30 and there are guards in forms of `or`s for cases when returned value from `_extract_first_element` is `None`

Reply:
Hmm, I didn't look the another branch behaviour code. This might still puzzle somebody who sees it for first time.

Reply:
It is possible yes. I was just contributing to the existing structure to respect the stylistic choices here. Would you like this code written differently to improve its readability?